ls1intum/Artemis
# `Plagiarism checks`: Fix formatting of similarity distribution

### Checklist
#### General


#### Client

### Motivation and Context


When changing the chart implementation from `chart.js` to `ngx-charts`, I implemented the y-axis formatting in a way, that does not scale for large maximal values.

### Description

Instead of always formatting the y-axis in integer steps, I declared a minimum y-intercept of 5. This means if only values < 5 are displayed by the chart, the highest value on the y-axis is 5. If at least one value is > 5, the y-axis adapts accordingly.
### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Programming/Text/Modeling Exercise with Plagiarisms

In general there are two possibilities to test these changes:

- Simulate plagiarisms on a test server: This might work, but only in small scale (you would have to do > 20 Submissions with some similarity in order to test these changes for large values)
- Use your local setup: Checkout this branch and use e.g, the following test data:
```
ngxTestData = [
        { name: '0%-10%', value: 0 },
        { name: '10%-20%', value: 0 },
        { name: '20%-30%', value: 0 },
        { name: '30%-40%', value: 0 },
        { name: '40%-50%', value: 13 },
        { name: '50%-60%', value: 24 },
        { name: '60%-70%', value: 36 },
        { name: '70%-80%', value: 41 },
        { name: '80%-90%', value: 57 },
        { name: '90%-100%', value: 0 },
    ];
```
Copy this and paste it as attribute in `plagiarism-run-details.component.ts`. Additionally, you have to change line 19 in the corresponding HTML template to: `[results]="ngxTestData"`

In either case, the testing steps are similar:

1. Log in to Artemis
2. Navigate to Course Administration
3. Navigate to the exercise details page of your prepared exercise
4. Click the "Check plagiarism" button
5. Run the plagiarism detection once
6. Click on "Run details"
7. Check the chart. The y axis should now be formatted in bigger steps for large values
     - If you work on your local setup: add `ngxTestData` as attribute in `plagiarism-run-details.component.ts` and change the corresponding line in the html template (described above). Feel free to change the test values in order to check how the chart looks if all values are below 10 or even 5 (or even smaller).

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage






### Screenshots


How it currently looks for large values:
![image (5)](https://user-images.githubusercontent.com/80622272/150341945-1ef48628-1934-4c08-a47b-7cd5edd57e14.png)

How it looks in this PR (different data set):
![image](https://user-images.githubusercontent.com/80622272/150342054-44da012f-c4e9-4813-8d89-85a36f42740e.png)
